### PR TITLE
Default allowed mentions

### PR DESCRIPTION
**Pull request by TheKodeToad**
## Checklist
- [x] I have tested this PR[^1].
- [x] I have read the [contributing guidelines](https://github.com/Javacord/Javacord/blob/master/.github/CONTRIBUTING.md).

## Changelog
- Added default allowed mentions to the `DiscordApi`

## Description
I used a slightly hacky approach of passing around DiscordApi. I hope it's okay!

Closes: #527

[^1]: At least started a running bot instance with your changes and triggered an event so your changed code gets executed.

### CODE REVIEW

**Comment by TheKodeToad** [comment]
I tested a bot but I didn't test `./gradlew build` :)

File: javacord-api/src/main/java/org/javacord/api/DiscordApiBuilder.java [hunk 1]
```diff
@@ -450,6 +452,28 @@ public DiscordApiBuilder setAllIntentsWhere(Predicate<Intent> condition) {
...
+
+    /**
+     * Controls who will be mentioned if mentions exist in a message.
+     * This is overriden with {@link MessageBuilder#setAllowedMentions(AllowedMentions)}
+     *
+     * @return The mention object.
+     */
+    public Optional<AllowedMentions> getAllowedMentions() {
+        return delegate.getAllowedMentions();
+    }
```
Comment (felldo) [inline javacord-api/src/main/java/org/javacord/api/DiscordApiBuilder.java:475]: We usually don't have getters in the `DiscordApiBuilder`
```suggestion
```

File: javacord-api/src/main/java/org/javacord/api/internal/DiscordApiBuilderDelegate.java [hunk 2]
```diff
@@ -177,6 +179,22 @@ public interface DiscordApiBuilderDelegate {
...
+    void setAllowedMentions(AllowedMentions allowedMentions);
+
+    /**
+     * Controls who will be mentioned if mentions exist in a message.
+     * This is overriden with {@link MessageBuilder#setAllowedMentions(AllowedMentions)}
+     *
+     * @return The mention object.
+     */
+    Optional<AllowedMentions> getAllowedMentions();
+
```
Comment (felldo) [inline javacord-api/src/main/java/org/javacord/api/internal/DiscordApiBuilderDelegate.java:197]:
```suggestion
```

**Review by felldo — CHANGES_REQUESTED** [review]
Looks good so far, but:
* Please rename every field/method/Javadoc so it says `defaultAllowedMentions`.
* You have added the DiscordApi to a few methods, please add it as the first parameter

File: javacord-api/src/main/java/org/javacord/api/DiscordApiBuilder.java (hunk 1 above)
Comment (TheKodeToad) [reply javacord-api/src/main/java/org/javacord/api/DiscordApiBuilder.java:475]: There are getters for e.g. bot token in the builder 🤔

Comment (felldo) [reply javacord-api/src/main/java/org/javacord/api/DiscordApiBuilder.java:475]: There are a few exceptions for like the bot token you mentioned or also the total shards. But if you look at the other 30 setters, none of them has getters. Not sure about the exact use case of the current getters, but I doubt getting the allowed mentioned on the builder instance is going to provide any benefits

**Comment by TheKodeToad** [comment]
Hmm, the constructor which doesn't use a websocket doesn't have a defaultAllowedMentions parameter... is that an issue? It's internal anyway, right?

**Comment by felldo** [comment]
> Hmm, the constructor which doesn't use a websocket doesn't have a defaultAllowedMentions parameter... is that an issue? It's internal anyway, right?

What do you mean? Can you show me which line?

**Comment by TheKodeToad** [comment]
> What do you mean? Can you show me which line?

https://github.com/TheKodeToad/Javacord/blob/default-allowed-mentions/javacord-core/src/main/java/org/javacord/core/DiscordApiImpl.java#L422

**Comment by felldo** [comment]
There are a lot of other things the constructor doesn't have, so it's okay

**Comment by TheKodeToad** [comment]
This PR should be ready then! hopefully I didn't miss anything...

**Review by felldo — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**